sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -561,16 +561,9 @@ case class TruncateTableCommand(
         }
       }
     }
-    // After deleting the data, invalidate the table to make sure we don't keep around a stale
-    // file relation in the metastore cache.
-    spark.sessionState.refreshTable(tableName.unquotedString)
-    // Also try to drop the contents of the table from the columnar cache
-    try {
-      spark.sharedState.cacheManager.uncacheQuery(spark.table(table.identifier), cascade = true)
-    } catch {
-      case NonFatal(e) =>
-        log.warn(s"Exception when attempting to uncache table $tableIdentWithDB", e)
-    }
+    // After deleting the data, refresh the table to make sure we don't keep around a stale
+    // file relation in the metastore cache and cached table data in the cache manager.
+    spark.catalog.refreshTable(tableIdentWithDB)
```

Comment:
Hmm... Not sure if we should catch any non-fatal exception like the previous one, since otherwise we'd skip updating stats?

Reply:
1. We don't catch any exception in other commands
2. What kind of exceptions should we hide from users here?

Reply:
Not clear which exceptions are caught here. `uncacheQuery()` doesn't throw anything, `spark.table(table.identifier).logicalPlan` could fail but in that case it is not clear how we reached this point.

Reply:
I think someone could drop a table or permanent view in a different session, or drop a Hive table through beeline or HMS API. This may cause some cache which depend on them AND this truncated table to become invalid, and potentially analysis exception when recaching them. I haven't got a chance to verify this though.

I feel overall it will be a good practice to recover from unknown errors here and continue. Also `DropTableCommand` does this as well.

Reply:
The case which you described here is applicable to any other commands like add/drop/rename/recover partitions. I do believe we either should "fix" all commands with tests for the case, or apply the approach w/o catching exceptions here as we do in other commands so far (otherwise the implementation looks inconsistent).

Reply:
Personally I'd keep just the try-catch logic here because I think the above do happens and we shouldn't skip updating stats in the case. But I don't really have strong opinion on this.

Reply:
can we fix the inconsistency first? i.e. reach an agreement about whether we should add `try-cache` or not for all other commands.

Reply:
> ... someone could drop a table or permanent view in a different session, or drop a Hive table through beeline or HMS API.

If somebody dropped the table in parallel, updating statistics wouldn't matter any more. We should show the error to user as soon as possible.

> can we fix the inconsistency first?

@sunchao Can you write a test which reproduces the issue?

Reply:
We might utilize `HiveThriftServer2Suites` for this - I can check when I got time.

Reply:
So I wasn't able to reproduce it with the above example, sorry for the false alarm. Turned out the analysis exception will be thrown later when the cache is actually queried (rather than in `recacheByPlan` itself). Therefore, I think it should be fine in this case.

I do agree we should keep it consistent (whether try-catch or not). IMO it can be done separately tho.